packages/flutter/lib/src/rendering/child_view.dart

```diff
@@ -0,0 +1,207 @@
+    assert(_viewToken == null);
+    mojom.ViewProviderProxy viewProvider = new mojom.ViewProviderProxy.unbound();
+    shell.connectToService(url, proxy);
+    mojom.ViewToken viewToken = (await viewProvider.ptr.createView(incommingServices, outgoingServices)).viewToken;
+    viewProvider.close();
+    assert(_viewToken == null);
+    _viewToken = viewToken;
+    assert(_viewKey == null);
+    if (_attached)
+      _addChildToViewHost();
+    return viewToken;
+  }
+
+  static int _nextViewKey = 1;
+  int _viewKey;
+
+  void _addChildToViewHost() {
+    assert(_attached);
+    assert(_viewToken != null);
+    assert(_viewKey == null);
+    _viewKey = _nextViewKey++;
+    _viewHost.addChild(_viewKey, _viewToken);
+  }
+
+  void _removeChildFromViewHost() {
+    assert(!_attached);
+    assert(_viewToken != null);
+    assert(_viewKey != null);
+    _viewHost.removeChild(_viewKey);
+    _viewKey = null;
+  }
+
+  // The number of render objects attached to this view. In between frames, we
+  // might have more than one connected if we get added to a new render object
+  // before we get removed from the old render object. By the time we get around
+  // to computing our layout, we must be back to just having one render object.
+  int _attachments = 0;
+  bool get _attached => _attachments > 0;
+
+  void _attach() {
+    assert(_attachments >= 0);
+    ++_attachments;
+    if (_viewToken != null && _viewKey == null)
+      _addChildToViewHost();
+  }
+
+  void _detach() {
+    assert(_attached);
+    --_attachments;
+    scheduleMicrotask(_removeChildFromViewHostIfNeeded);
+  }
+
+  void _removeChildFromViewHostIfNeeded() {
+    assert(_attachments >= 0);
+    if (_attachments == 0)
+      _removeChildFromViewHost();
+  }
+
+  Future<mojom.ViewLayoutInfo> _layout({ Size size, double scale }) async {
+    assert(_attached);
+    assert(_attachments == 1);
+    assert(_viewKey != null);
+    int width = (size.width * scale).round();
+    int height = (size.height * scale).round();
+    mojom.BoxConstraints childConstraints = new mojom.BoxConstraints()
+      ..minWidth = width
+      ..maxWidth = width
+      ..minHeight = height
+      ..maxHeight = height;
+    mojom.ViewLayoutParams layoutParams = new mojom.ViewLayoutParams()
+      ..constraints = childConstraints
+      ..devicePixelRatio = scale;
+    return (await _viewHost.layoutChild(_viewKey, layoutParams)).info;
+  }
+
+  String toString() {
+    return '$runtimeType(url: $url)';
+  }
+}
+
+class RenderChildView extends RenderBox {
+  RenderChildView({
+    ChildViewConnection child,
+    double scale
+  }) : _child = child, _scale = scale;
+
+  ChildViewConnection get child => _child;
+  ChildViewConnection _child;
+  void set child (ChildViewConnection value) {
+    if (value == _child)
+      return;
+    if (attached)
+      _child?._detach();
+    _child = value;
+    _layoutInfo = null;
+    if (attached)
+      _child._attach();
```

Comment:
I went back and forth on this point (hence the confusion in the code).  I think it makes sense for child to be null, so I'll add the `?.`.